crates/aptos/src/common/init.rs

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    common::{
+        types::CliConfig,
+        utils::{prompt_yes, to_common_success_result},
+    },
+    op::key::GenerateKey,
+    CliResult, Error,
+};
+use aptos_crypto::{x25519::PrivateKey, ValidCryptoMaterialStringExt};
+use clap::Parser;
+
+/// Tool to initialize current directory for the aptos tool
+#[derive(Debug, Parser)]
+pub struct InitTool {}
+
+impl InitTool {
+    pub async fn execute(&self) -> CliResult {
+        to_common_success_result(self.execute_inner().await)
+    }
+
+    async fn execute_inner(&self) -> Result<(), Error> {
+        let mut config = if CliConfig::config_exists()? {
+            if !prompt_yes(
+                "Aptos already initialized, do you want to overwrite the existing config?",
+            ) {
+                eprintln!("Exiting...");
+                return Ok(());
+            }
+            CliConfig::load()?
+        } else {
+            CliConfig::default()
+        };
+
+        eprintln!("Enter your private key as a hex literal (0x...) [No input: Generate new key]");
+        let input = read_line()?;
+        let input = input.trim();
+        let private_key = if input.is_empty() {
```

Comment:
this could be a match

Reply:
I'm not sure it'll be more readable though.  I'll leave it like this for now.